api/v1alpha3/awscluster_types.go

```diff
@@ -107,6 +107,10 @@ type AWSLoadBalancerSpec struct {
 	// Defaults to false.
 	// +optional
 	CrossZoneLoadBalancing bool `json:"crossZoneLoadBalancing,omitempty"`
+
+	// Subnets specifies the subnets that should be used by the load balancer
+	// +optional
+	Subnets Subnets `json:"subnets,omitempty"`
```

Comment:
We should scope this down to the bits that we're actually using, otherwise the API is going to be problematic, as it includes references to NAT gateways and public and/or private subnets. Copying the types to be more local to the task in hand is fine.

Reply:
It might make more sense to just add the same `subnetIds` array that's in status up into the spec section.

Reply:
Since we're adding ability to specify subnets with filters with #1776, can make it a slice of AWSResourceReference